### ISSUE TITLE
level 1 of method

### ISSUE THREAD

**Issue by shawncch** [nus-cs2030/2223-s2#621]
hi, may i know if we are allowed to use instanceof in our of method to check if t is a log?

i'm currently using a filter -> map -> orElseThrow for my of(T t) method, haven't thought of how to check both T and Log for the other of method

**Comment by leezhanpeng** [comment]
Yes, `instanceof` is allowed in this lab.

**Comment by siruislybro** [comment]
Regarding the second part, I believe you are only allowed to use` orElseThrow` and `IllegalArgumentException `once, which means that the checking of null values and instanceof log can only be done in one of the `of` methods

**Comment by briannoelkesuma** [comment]
yes you can use instanceof. u can try filtering out instanceof Log for typechecking purposes!

**Comment by EvanCh08** [comment]
I believe you also can use Optional.ofNullable() -> filter -> flatMap -> exception

**Comment by koiljat** [comment]
you can check for Log using instanceof using filter, then use flatMap to check for null values for both

**Comment by davytan07** [comment]
Can refer to [this thread](https://github.com/nus-cs2030/2223-s2/issues/587#issuecomment-1483985343)! Basically the idea is to chain filters and map/flatMap

**Comment by mkim123** [comment]
yes, `instanceof` is allowed

**Comment by e0968868** [comment]
Yes you can use it! its not one of the restrictions

**Comment by gene2345** [comment]
yes u can use instanceof

**Comment by haqanh** [comment]
Yep, you can use instanceof in .equals() method. and you can use, as hinted in the Lab, ofNullable() then make use of map/flatMap for this of() method.
Hope this helps!

**Comment by user1914** [comment]
yes you are allowed to use instanceof for this lab to check for whether the input is of type Log
you can try using Optional.ofNullable twice to check

**Comment by ouhznay** [comment]
you can try returning the 2 input of method so that you dont have to check twice
